Changed mapping for geo_point to npgsql point datatype (oid 600)

mschwoebel opened the pull request:
Fixed and extended unit tests

Please verify correct mapping for point. 
Before it was double[] and now to Npgsql datatype point

mschwoebel commented:
@mfussenegger thx for the fast check. I added it to the change log. Please check if this meets your conventions. If everything is ok i will merge it to dev branch tomorrow.

mfussenegger commented on CHANGES.rst:
```diff
@@ -5,6 +5,11 @@ Changes in Npgsql-Plugin for Crate
+2019/08/24 1.2.4
+================
+
+- Breaking change: Changed mapping for geo_point to npgsql point datatype (oid 600)
```
Maybe we could clarify that this change only has an effect if the CrateDB nodes are >= 4.1

```suggestion
-  Added a type mapping from ``geo_point`` (oid 600) to ``NpgsqlPoint``, to be
   able to handle ``geo_point`` values with CrateDB >= 4.1. There the streaming
   changed, so that ``geo_point`` values are no longer streamed as double array
   but as ``geo_point``.
```

mschwoebel replied:
ok thx thats sounds good. One more question.. should the new version of the npgsql-crate plugin support crate version >= 4.1 and crate version <4.1? I did a small change to support both, but before i commit it can you please give me your opinion. thx

mfussenegger replied:
Yes ideally it would support both

mschwoebel commented:
i move the changes to the unreleased part and updated code to support crate version >= 4.1 and less than 4.1.
I added missing unit test for geo_point array data type too.

mfussenegger commented on src/Npgsql.CrateDb/CrateDbDatabaseInfo.cs:
```diff
@@ -71,7 +71,16 @@ public CrateDbDatabaseInfo(NpgsqlConnection conn)
+            //but as 'geo_point'.
+            if (Version >= new Version("4.1.0"))
+            {
+                CrateDbBaseTypes.Add("point", 600);
```
Isn't it possible to always add this, even for CrateDB <= 4.0? 

The server communicates the oid's to the client. in CrateDB 4.0 or earlier it will simply never use `600`, but `1022` (double array)

mschwoebel replied:
yeah you are right. i will add it to the base mapping. Thx for the info. 
Commit will follow in few minutes

mfussenegger commented:
@mschwoebel thank you, I've squashed the commits and merged it now.
